### PR TITLE
implement standalone functionality of FoamDictParser

### LINKED ISSUE #21: Owls2.0 FoamDictParser doesn't work as standalone
Currently, `Owls.parser.FoamDict.FileParser` do only work as mixins for example as used in [OBR](https://github.com/hpsim/OBR/blob/77685d9b7669823177d223356cda26070b01cb00/src/obr/OpenFOAM/case.py#LL94C26-L94C88 ). This should be refactored so one can do something like:

```
  from Owls.parser.FoamDict import FileParser
  
  fp = FileParser(path="path")
  fp.get("key")
```
  
I guess when using `FileParser` as mixin for something else `FileParser.path` can just be overwritten.

### PR DESCRIPTION
**Pull request by lupeterm**
also, fixes some pyparsing naming errors, e.g., chenging `set_results_name` to `setResultsName`.
fixes #21

### CODE REVIEW

File: src/Owls/parser/FoamDict.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def parse():
             + pp.Suppress("]")
             + pp.Word(pp.alphanums + '_-."/')
             + pp.Suppress(";")
-        ).set_results_name("of_dimension_set")
+        ).setResultsName("of_dimension_set")
```
Comment (greole) [inline src/Owls/parser/FoamDict.py]: I think the changed it the other way around, setResultsName is deprecated in favour of set_results_name

Comment (greole) [reply src/Owls/parser/FoamDict.py]: See https://pyparsing-docs.readthedocs.io/en/latest/pyparsing.html?highlight=setResultsName#pyparsing.ParserElement.setResultsName

File: src/Owls/parser/FoamDict.py [hunk 2]
```diff
@@ -40,10 +40,10 @@ def parse():
         return (
             pp.Word(pp.alphanums)
             + pp.Suppress("(")
-            + pp.OneOrMore(pp.Word(pp.alphanums + '_-."/'), stop_on=pp.Literal(")"))
+            + pp.OneOrMore(pp.Word(pp.alphanums + '_-."/'), stopOn=pp.Literal(")"))
```
Comment (greole) [inline src/Owls/parser/FoamDict.py]: I think pyparsing switched from camelCase to snake_case, see https://pyparsing-docs.readthedocs.io/en/latest/whats_new_in_3_0_0.html#pep-8-naming

**Review by greole — CHANGES_REQUESTED** [review]
Hey @lupeterm, thanks for your PR. I think that goes into the right direction. Could you  

1. Just revert back to `snake_case` for pyparsing as it seems to be the newer syntax.
2. Add a unit test to [owls/test//test_dictParser.py](https://github.com/greole/owls/tree/Owls2.0/tests/test_dictParser.py)
 for the functionality. Maybe by just take this [controlDict](https://github.com/OpenFOAM/OpenFOAM-10/blob/master/tutorials/incompressible/simpleFoam/airFoil2D/system/controlDict) as an example and write it to a `tmpDir` fixture

File: src/Owls/parser/FoamDict.py (hunk 1 above)
Comment (lupeterm) [reply src/Owls/parser/FoamDict.py]: Thanks @greole, i reverted my changes back to the non-deprecated version. Somehow, the snake-case spelling threw a couple errors the first time.

File: src/Owls/parser/FoamDict.py [hunk 3]
```diff
@@ -198,7 +205,7 @@ def key_value_to_dict(self, parse_result):
                         d = {key: self.key_value_to_dict([v for v in res.value])}
                         ret.update(d)
                     elif res.get("of_list"):
-                        tmp_list = res.get("of_list").as_list()
+                        tmp_list = res.get("of_list").asList()
```
Comment (greole) [inline src/Owls/parser/FoamDict.py]: keep it as_list()
```suggestion
                        tmp_list = res.get("of_list").as_list()
```

File: src/Owls/parser/FoamDict.py [hunk 4]
```diff
@@ -181,8 +188,8 @@ def key_value_to_dict(self, parse_result):
         ret = {}
         for res in parse_result:
             # probe if next result is str or ParseResult
-            if isinstance(res, pp.results.ParseResults):
-                if res.get_name() == "key_value_pair":
+            if isinstance(res, pp.ParseResults):
+                if res.getName() == "key_value_pair":
```
Comment (greole) [inline src/Owls/parser/FoamDict.py]: keep snake case?
```suggestion
                if res.get_name() == "key_value_pair":
```

File: src/Owls/parser/FoamDict.py [hunk 5]
```diff
@@ -168,7 +175,7 @@ def config_parser(self):
             pp.ZeroOrMore(self.single_line_comment) ^ pp.ZeroOrMore(self.key_value_pair)
         )
 
-    def convert_to_number(self, inp: list[str]):
+    def convert_to_number(self, inp):
```
Comment (greole) [inline src/Owls/parser/FoamDict.py]: why do we remove the type hint here?

File: src/Owls/parser/FoamDict.py [hunk 6]
```diff
@@ -232,7 +239,7 @@ def parse_file_to_dict(self):
 
     def parse_str_to_dict(self, s) -> dict:
         """Parse a given FoamDict body str to a python dictionary"""
-        self.parse = self.config_parser.search_string(s)
+        self.parse = self.config_parser.searchString(s)
```
Comment (greole) [inline src/Owls/parser/FoamDict.py]: keep snake case?
```suggestion
        self.parse = self.config_parser.search_string(s)
```

File: src/Owls/parser/FoamDict.py [hunk 7]
```diff
@@ -123,6 +124,12 @@ def footer(self):
     def separator(self):
         return "// " + "* " * 26 + "//"
 
+    def get(self, key):
+        try:
+            return self._dict[key]
+        except KeyError as e:
+            print(e)
+
```
Comment (greole) [inline src/Owls/parser/FoamDict.py]:
```suggestion
    def get(self, key, default=None):
            return self._dict.get(key, default)
```
I think it is better to just forward get to self._dict

File: src/Owls/parser/FoamDict.py (hunk 5 above)
Comment (lupeterm) [reply src/Owls/parser/FoamDict.py]: I pushed it by mistake. Iirc, up until Python 3.10, the `list` annotation needed to be imported like `from typing import List`.  I only recently upgraded to 3.10.

File: src/Owls/parser/FoamDict.py (hunk 7 above)
Comment (lupeterm) [reply src/Owls/parser/FoamDict.py]: I did not know there was a default option for `dict.get()`. Good to know, thanks!

**Pull request merged**